crates/router/src/cache.rs

```diff
@@ -1,23 +1,69 @@
-use std::{any::Any, sync::Arc};
+use std::{any::Any, borrow::Cow, sync::Arc};
 
 use dyn_clone::DynClone;
+use error_stack::Report;
 use moka::future::Cache as MokaCache;
 use once_cell::sync::Lazy;
+use redis_interface::RedisValue;
+
+use crate::core::errors;
 
 /// Time to live 30 mins
 const CACHE_TTL: u64 = 30 * 60;
 
 /// Time to idle 10 mins
 const CACHE_TTI: u64 = 10 * 60;
 
+/// Max Capacity of Cache in MB
+const MAX_CAPACITY: u64 = 30;
+
 /// Config Cache with time_to_live as 30 mins and time_to_idle as 10 mins.
-pub static CONFIG_CACHE: Lazy<Cache> = Lazy::new(|| Cache::new(CACHE_TTL, CACHE_TTI));
+pub static CONFIG_CACHE: Lazy<Cache> = Lazy::new(|| Cache::new(CACHE_TTL, CACHE_TTI, None));
+
+/// Accounts cache with time_to_live as 30 mins and size limit
+pub static ACCOUNTS_CACHE: Lazy<Cache> =
+    Lazy::new(|| Cache::new(CACHE_TTL, CACHE_TTI, Some(MAX_CAPACITY)));
 
 /// Trait which defines the behaviour of types that's gonna be stored in Cache
 pub trait Cacheable: Any + Send + Sync + DynClone {
     fn as_any(&self) -> &dyn Any;
 }
 
+pub enum CacheKind<'a> {
+    Config(Cow<'a, str>),
+    Accounts(Cow<'a, str>),
+}
+
+impl<'a> From<CacheKind<'a>> for RedisValue {
+    fn from(kind: CacheKind<'a>) -> Self {
+        let value = match kind {
+            CacheKind::Config(s) => format!("config,{s}"),
+            CacheKind::Accounts(s) => format!("accounts,{s}"),
+        };
+        Self::String(value.into())
+    }
+}
+
+impl<'a> TryFrom<RedisValue> for CacheKind<'a> {
+    type Error = Report<errors::ValidationError>;
+    fn try_from(kind: RedisValue) -> Result<Self, Self::Error> {
+        let validation_err = errors::ValidationError::InvalidValue {
+            message: "Invalid publish key provided in pubsub".into(),
+        };
+        let kind = kind.as_string().ok_or(validation_err.clone())?;
+        let mut split = kind.split(',');
+        match split.next().ok_or(validation_err.clone())? {
+            "accounts" => Ok(Self::Accounts(Cow::Owned(
+                split.next().ok_or(validation_err)?.to_string(),
+            ))),
+            "config" => Ok(Self::Config(Cow::Owned(
+                split.next().ok_or(validation_err)?.to_string(),
+            ))),
```

Comment:
Would be preferable if these keys were `const`s instead of static string literals repeated multiple times.